RecoBTag/FeatureTools/plugins/HiggsInteractionNetTagInfoProducer.cc

```diff
@@ -0,0 +1,353 @@
+      fts.fill("cpf_quality", packed_cand->pseudoTrack().qualityMask());
+
+      // impact parameters
+      fts.fill("cpf_dz", catch_infs(packed_cand->dz()));
+      fts.fill("cpf_dzsig", catch_infs_and_bound(packed_cand->dz() / packed_cand->dzError(), 0, -2000, 2000));
+      fts.fill("cpf_dxy", catch_infs(packed_cand->dxy()));
+      fts.fill("cpf_dxysig", catch_infs_and_bound(packed_cand->dxy() / packed_cand->dxyError(), 0, -2000, 2000));
+    } else if (reco_cand) {
+      fts.fill("cpf_quality", quality_from_pfcand(*reco_cand));
+
+      // impact parameters
+      const auto *trk = reco_cand->bestTrack();
+      float dz = trk->dz(pv_->position());
+      float dxy = trk->dxy(pv_->position());
+      fts.fill("cpf_dz", catch_infs(dz));
+      fts.fill("cpf_dzsig", catch_infs_and_bound(dz / trk->dzError(), 0, -2000, 2000));
+      fts.fill("cpf_dxy", catch_infs(dxy));
+      fts.fill("cpf_dxysig", catch_infs_and_bound(dxy / trk->dxyError(), 0, -2000, 2000));
+    }
+
+    // basic kinematics
+    fts.fill("cpf_phirel", reco::deltaPhi(candP4, jet));
+    fts.fill("cpf_etarel", etasign * (candP4.eta() - jet.eta()));
+    fts.fill("cpf_deltaR", reco::deltaR(candP4, jet));
+
+    fts.fill("cpf_ptrel", catch_infs(candP4.pt() / jet.pt(), -99));
+    fts.fill("cpf_erel", catch_infs(candP4.energy() / jet.energy(), -99));
+
+    float drminpfcandsv = btagbtvdeep::mindrsvpfcand(*svs_, &(*cand), jet_radius_);
+    fts.fill("cpf_drminsv", catch_infs_and_bound(drminpfcandsv, 0, -1. * jet_radius_, 0, -1. * jet_radius_));
+
+    // subjets
+    auto subjets = patJet->subjets();
+    std::sort(subjets.begin(), subjets.end(), [](const edm::Ptr<pat::Jet> &p1, const edm::Ptr<pat::Jet> &p2) {
+      return p1->pt() > p2->pt();
+    });  // sort by pt
+    fts.fill("cpf_drsubjet1", !subjets.empty() ? reco::deltaR(*cand, *subjets.at(0)) : -1);
+    fts.fill("cpf_drsubjet2", subjets.size() > 1 ? reco::deltaR(*cand, *subjets.at(1)) : -1);
+
+    const reco::Track *trk = nullptr;
+    if (packed_cand && packed_cand->hasTrackDetails()) {
+      trk = &(packed_cand->pseudoTrack());
+    } else if (reco_cand && useTrackProperties(reco_cand)) {
+      trk = reco_cand->bestTrack();
+    }
+    TrackInfoBuilder trkinfo(track_builder_);
+    trkinfo.buildTrackInfo(&(*cand), jet_dir, jet_ref_track_dir, *pv_);
+    fts.fill("cpf_btagEtaRel", catch_infs_and_bound(trkinfo.getTrackEtaRel(), 0, -5, 15));
+    fts.fill("cpf_btagPtRatio", catch_infs_and_bound(trkinfo.getTrackPtRatio(), 0, -1, 10));
+    fts.fill("cpf_btagPParRatio", catch_infs_and_bound(trkinfo.getTrackPParRatio(), 0, -10, 100));
+    fts.fill("cpf_btagSip2dVal", catch_infs_and_bound(trkinfo.getTrackSip2dVal(), 0, -1, 70));
+    fts.fill("cpf_btagSip2dSig", catch_infs_and_bound(trkinfo.getTrackSip2dSig(), 0, -1, 4e4));
+    fts.fill("cpf_btagSip3dVal", catch_infs_and_bound(trkinfo.getTrackSip3dVal(), 0, -1, 1e5));
+    fts.fill("cpf_btagSip3dSig", catch_infs_and_bound(trkinfo.getTrackSip3dSig(), 0, -1, 4e4));
```

Comment:
individual track significances are clamped at 2e3, would the same value work here?
Again, trying to reduce the entropy of random numbers in the code.